Traceback (most recent call last):
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\asyncio\base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\gabri\miniconda3\envs\programacioi\Lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
c1 = Complex(2,3)
c2 = Complex(3,4)

c1.real = "matematiques" # canviam el tipus d'una variable de enter a string

c3 = c1.suma(c2) # aixo dona error ja que no pot sumar un string amb un enter
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mTypeError[0m                                 Traceback (most recent call last)
Cell [1;32mIn[9], line 6[0m
[0;32m      2[0m c2 [38;5;241m=[39m Complex([38;5;241m3[39m,[38;5;241m4[39m)
[0;32m      4[0m c1[38;5;241m.[39mreal [38;5;241m=[39m [38;5;124m"[39m[38;5;124mmatematiques[39m[38;5;124m"[39m [38;5;66;03m# canviam el tipus d'una variable de enter a string[39;00m
[1;32m----> 6[0m c3 [38;5;241m=[39m c1[38;5;241m.[39msuma(c2)

Cell [1;32mIn[8], line 10[0m, in [0;36mComplex.suma[1;34m(self, o)[0m
[0;32m      9[0m [38;5;28;01mdef[39;00m [38;5;21msuma[39m([38;5;28mself[39m, o):    
[1;32m---> 10[0m     r [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mreal [38;5;241m+[39m o[38;5;241m.[39mreal
[0;32m     11[0m     i [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mimaginaria [38;5;241m+[39m o[38;5;241m.[39mimaginaria
[0;32m     12[0m     [38;5;28;01mreturn[39;00m Complex(r, i)

[1;31mTypeError[0m: can only concatenate str (not "int") to str
TypeError: can only concatenate str (not "int") to str

